File: benchmarks/stage5_vexp_swe_bench_smoke/fixtures/m202_language_families/haskell/valid.hs
-- Fixture — café 日本語 before declarations
module Fixture where

import Data.List (sort)

-- | A point.
data Point = Point { px :: Int, py :: Int }

newtype Wrapper = Wrapper Int

type Identifier = String

-- | A shape class.
class Shape a where
  area :: a -> Double

-- | Add two numbers.
add :: Int -> Int -> Int
add a b = a + b

helper :: [Int] -> [Int]
helper xs = sort xs
helper [] = []

File: benchmarks/stage5_vexp_swe_bench_smoke/fixtures/m202_language_families/clojure/valid.clj
;; Fixture — café 日本語 before declarations
(ns fixture.core
  (:require [clojure.string :as str]))

;; Add two numbers.
(defn add [a b]
  (+ a b))

(def limit 10)

(defmacro shout [text] `(str/upper-case ~text))

(defprotocol Shape
  (area [this]))

(defrecord Point [x y])

(defn- helper [x] (inc x))
